### PR TITLE
[type: refactor] refactor shenyu client.

### PR DESCRIPTION
**Pull request by qicz**
- replace magic value with constant; 
- reset parameter sort; 
- add ShenyuClientException; 
- fix SpringCloud Client path bug.

### CODE REVIEW

File: shenyu-client/shenyu-client-common/src/main/java/org/apache/shenyu/client/common/exception/ShenyuClientException.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.client.common.exception;
+
+/**
+ * ShenyuClientException.
+ */
+public class ShenyuClientException extends IllegalArgumentException {
```
Comment (tydhot) [inline shenyu-client/shenyu-client-common/src/main/java/org/apache/shenyu/client/common/exception/ShenyuClientException.java:23]: why it extend IllegalArgumentException?

Comment (qicz) [reply shenyu-client/shenyu-client-common/src/main/java/org/apache/shenyu/client/common/exception/ShenyuClientException.java:23]: I see the client just validating the arguments.

Comment (qicz) [reply shenyu-client/shenyu-client-common/src/main/java/org/apache/shenyu/client/common/exception/ShenyuClientException.java:23]: I see the client just validate the arguments.

Comment (tydhot) [reply shenyu-client/shenyu-client-common/src/main/java/org/apache/shenyu/client/common/exception/ShenyuClientException.java:23]: so why not change the name to `ShenyuIllegalException` ? And I think you can move them in `shenyu-client-core`, no need to creaye a new module like client-common.

**Review by tydhot — CHANGES_REQUESTED** [review]
I think we already have `shenyu-client-core`, no need to create a new module just put a class about contants and a class about exception

File: shenyu-client/shenyu-client-common/src/main/java/org/apache/shenyu/client/common/exception/ShenyuClientException.java (hunk 1 above)
Comment (qicz) [reply shenyu-client/shenyu-client-common/src/main/java/org/apache/shenyu/client/common/exception/ShenyuClientException.java]: did it. use `ShenyuClientIllegalArgumentException`.

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java [hunk 2]
```diff
@@ -38,16 +38,16 @@
...
-     * @param env the env
-     * @param shenyuClientRegisterRepository the shenyu client register repository
      * @param clientConfig the client config
+     * @param shenyuClientRegisterRepository the shenyu client register repository
+     * @param env the env
      * @return the spring cloud client bean post processor
      */
     @Bean
-    public SpringCloudClientBeanPostProcessor springCloudClientBeanPostProcessor(final Environment env,
+    public SpringCloudClientBeanPostProcessor springCloudClientBeanPostProcessor(final ShenyuClientConfig clientConfig,
```
Comment (tydhot) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java:47]: to be honset, i really think the change like this is meaningless

**Review by tydhot — APPROVED** [review]

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java (hunk 2 above)
Comment (qicz) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springcloud/src/main/java/org/apache/shenyu/springboot/starter/client/springcloud/ShenyuSpringCloudClientConfiguration.java:47]: just same like another clients.

**Pull request merged**